### PR TITLE
Display plugin version on home page

### LINKED ISSUE #362: Show version on landing page
It would be helpful if the design library would show its release version somewhere on the landing page. 

When I navigate to https://weekly.ci.jenkins.io/design-library/ I cannot see if the changes that are shown in https://github.com/jenkinsci/design-library-plugin/releases/ are already deployed or not.

### PR DESCRIPTION
**Pull request by krisstern**
### Description
Fixes #362
To follow up on #365, which was closed prematurely by contributor
Attributing credit to @SohamJuneja

### Testing done
Ran `mvn clean verify` on Java JDK version 21 to confirm all tests passed. 

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by krisstern** [comment]
@SohamJuneja If you want to continue work on this issue/PR, you can pull from my branch and start another PR, after which I will close this pull request. But do let me know if that is the case.

**Comment by janfaracik** [comment]
Hey big thanks all. I like the idea, I'd rather it not be so prominent though - how is something like this:

image

In `previous-next-controls.jelly`:

```xml
      <j:if test="${attrs.previousPage == null and isHome}">
        <a href="https://github.com/jenkinsci/design-library-plugin/releases/tag/${it.pluginVersion}" target="_blank" class="jenkins-button jenkins-button--tertiary">
          <l:icon src="symbol-design-library plugin-design-library" />
          <span>${it.pluginVersion}</span>
        </a>
      </j:if>
```

File: src/main/java/io/jenkins/plugins/designlibrary/Home.java [hunk 1]
```diff
@@ -45,4 +47,15 @@ public UISample getDynamic(String name) {
         }
         return null;
     }
+
+    public String getPluginVersion() {
+        Jenkins jenkins = Jenkins.get();
+        if (jenkins != null && jenkins.getPluginManager() != null) {
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/designlibrary/Home.java:53]: While Jenkins can be `null`, the plugin manager can't. 
```suggestion
        if (jenkins != null) {
```

**Review by uhafner — CHANGES_REQUESTED** [review]
I have no opinion where to place the version. Using the footer as recommended by @janfaracik is fine for me as well!

File: src/main/java/io/jenkins/plugins/designlibrary/Home.java (hunk 1 above)
Comment (SohamJuneja) [reply src/main/java/io/jenkins/plugins/designlibrary/Home.java:53]: I think you meant to remove jenkins!=null and let the other remain the same , right?

Comment (uhafner) [reply src/main/java/io/jenkins/plugins/designlibrary/Home.java:53]: No, `jenkins` can be `null`. 

`jenkins.getPluginManager()` != `null`

Comment (SohamJuneja) [reply src/main/java/io/jenkins/plugins/designlibrary/Home.java:53]: Okay , got it

**Comment by SohamJuneja** [comment]
I've made the desired changes to add this into footer. But I am facing a repeated error when using mvn clean install. Already shared that on gitter newcomer contributor channel , I can share a draft PR until then or can wait to get insights on that error

**Comment by krisstern** [comment]
> I've made the desired changes to add this into footer. But I am facing a repeated error when using mvn clean install. Already shared that on gitter newcomer contributor channel , I can share a draft PR until then or can wait to get insights on that error

Yeah, agreeing with Jan that you should first create a new PR with the latest changes, so we can check for you on our side to troubleshoot.

**Comment by SohamJuneja** [comment]
Added a draft PR , so this can be closed now. Thank you all for your guidance.

**Comment by krisstern** [comment]
Thanks for letting us know @SohamJuneja

**Pull request closed without merging**